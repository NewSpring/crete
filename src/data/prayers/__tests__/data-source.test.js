import PrayerDataSource from '../data-source';
import {
  threeSortedPrayers,
  threeUnsortedPrayers,
  menuCategories,
} from '../../mocks/prayer';

describe('PrayerRequest data sources', () => {
  let Prayer;
  beforeEach(() => {
    Prayer = new PrayerDataSource();
    Prayer.context = { dataSources: {} };
  });
  it('sorts prayers by count and date', () => {
    expect(threeSortedPrayers).toEqual(
      Prayer.sortPrayers(threeUnsortedPrayers)
    );
  });
  it('gets prayer menu categories', async () => {
    Prayer.request = () => ({
      filter: () => ({
        orderBy: () => ({
          get: jest.fn(() => menuCategories),
        }),
      }),
    });
    Prayer.context.dataSources.Auth = {
      getCurrentPerson: () => ({ id: 'Person123' }),
    };
    Prayer.context.dataSources.Campus = {
      getForPerson: () => ({ name: 'Web' }),
    };
    expect(await Prayer.getPrayerMenuCategories()).toMatchSnapshot();

    Prayer.context.dataSources.Campus = {
      getForPerson: () => ({ name: 'Greenville' }),
    };
    expect(await Prayer.getPrayerMenuCategories()).toMatchSnapshot();
  });
  it('gets an empty list of saved prayers', async () => {
    Prayer.context.dataSources.Followings = {
      getFollowingsForCurrentUser: () => ({ get: () => [] }),
    };
    const cursor = await Prayer.bySaved();
    expect(await cursor.get()).toMatchSnapshot();
    expect(await cursor.count()).toMatchSnapshot();
  });
  it('answers a prayer', async () => {
    Prayer.patch = () => null;
    Prayer.getFromId = () => ({ id: 1, answer: 'answer' });
    expect(await Prayer.editAnswer(1, 'answer')).toMatchSnapshot();
  });
  it('removes a prayer answer', async () => {
    Prayer.patch = () => null;
    Prayer.getFromId = () => ({ id: 1 });
    expect(await Prayer.editAnswer(1)).toMatchSnapshot();
  });
});
