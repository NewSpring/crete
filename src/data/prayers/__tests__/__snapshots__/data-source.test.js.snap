// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrayerRequest data sources gets prayer menu categories 1`] = `
Array [
  Object {
    "attributeValues": Object {
      "requiresCampusMembership": Object {
        "value": "False",
      },
    },
    "id": 2,
    "key": "community",
  },
]
`;

exports[`PrayerRequest data sources gets prayer menu categories 2`] = `
Array [
  Object {
    "attributeValues": Object {
      "requiresCampusMembership": Object {
        "value": "True",
      },
    },
    "id": 1,
    "key": "campus",
  },
  Object {
    "attributeValues": Object {
      "requiresCampusMembership": Object {
        "value": "False",
      },
    },
    "id": 2,
    "key": "community",
  },
]
`;
